<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2021-06-09" />

<title>Getting Started with Capture Hi-C Analysis Engine</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Getting Started with <strong>C</strong>apture <strong>Hi-C</strong> <strong>An</strong>alysis <strong>E</strong>ngine</h1>
<h4 class="date">2021-06-09</h4>


<div id="TOC">
<ul>
<li><a href="#background">Background</a></li>
<li><a href="#quickstart">Quickstart</a></li>
<li><a href="#preprocessing">Pre-processing data</a></li>
<li><a href="#statistical-model">Statistical model</a><ul>
<li><a href="#distribution-of-the-counts">Distribution of the counts</a><ul>
<li><a href="#negative-binomial">Negative binomial</a></li>
<li><a href="#poisson">Poisson</a></li>
<li><a href="#truncated-negative-binomial">Truncated negative binomial</a></li>
<li><a href="#truncated-poisson">Truncated poisson</a></li>
</ul></li>
<li><a href="#adjusting-for-covariates">Adjusting for covariates</a><ul>
<li><a href="#custom-covariates">Custom covariates</a></li>
</ul></li>
<li><a href="#filtering-baits-and-targets">Filtering baits and targets</a></li>
<li><a href="#merging-biological-replicates">Merging biological replicates</a></li>
<li><a href="#assessing-model-fits">Assessing model fits</a></li>
<li><a href="#setting-distance-bins">Setting distance bins</a></li>
<li><a href="#multiple-testing-correction">Multiple testing correction</a></li>
</ul></li>
<li><a href="#interpretation-of-interaction-peaks">Interpretation of interaction peaks</a><ul>
<li><a href="#annotation-with-bedtools">Annotation with bedtools</a></li>
<li><a href="#visualising-interaction-peaks-in-washu-epigenome-browser">Visualising interaction peaks in WashU Epigenome Browser</a></li>
<li><a href="#locus-plot">Locus plots</a></li>
<li><a href="#background-interactions">Background Interactions</a></li>
</ul></li>
<li><a href="#example-workflow-t-47d-breast-cancer-cell-line">Example Workflow (T-47D breast cancer cell line)</a></li>
<li><a href="#session-info">Session Info</a></li>
<li><a href="#acknowledgements">Acknowledgements</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="background" class="section level1">
<h1>Background</h1>
<p>CHiCANE is a toolkit for identifying chromatin regions that interact more often than expected by chance given the specific interaction dynamics of a capture Hi-C (CHi-C) experiment. It offers extensive flexibility in implementing interaction calling via numerous optional parameters so that it can be tailored to a wide range of CHi-C experiments. Additionally, CHiCANE includes multiple helper functions for use in both pre- and post-processing/analysis of the interaction calls.</p>
<p>For details of statistical concepts implemented by CHiCANE and their application to both the region CHi-C and promoter CHi-C datasets, please see our manuscript: Holgerson, Erle, Gillespie, Andrea, et al. “Identifying High Confidence Capture Hi-C Interactions Using CHiCANE” (in revision 2020).</p>
</div>
<div id="quickstart" class="section level1">
<h1>Quickstart</h1>
<p>The main function for running the CHiCANE method is <code>chicane()</code>. This takes a bam file and bed files with all restriction fragments and targeted restriction fragments as input, and produces a table with fragment interactions and associated p-values.</p>
<p>The package comes with a subset of reads from a capture Hi-C experiment on the Bre80 normal epithelial breast tissue cell line <span class="citation">(Baxter et al. 2018)</span>. The experiment targeted several breast cancer risk loci in non-coding regions of the genome, and aimed to associate these risk variants with genes. Two of the risk SNPs are rs13387042 and rs16857609, and reads that map to them have been included in the file Bre80_2q35.bam.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(chicane);</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co">#&gt; Loading required package: gamlss.tr</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co">#&gt; Loading required package: gamlss.dist</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co">#&gt; Loading required package: MASS</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co">#&gt; Loading required package: gamlss</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co">#&gt; Loading required package: splines</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co">#&gt; Loading required package: gamlss.data</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co">#&gt; </span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co">#&gt; Attaching package: &#39;gamlss.data&#39;</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co">#&gt; The following object is masked from &#39;package:datasets&#39;:</span></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="co">#&gt; </span></span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="co">#&gt;     sleep</span></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="co">#&gt; Loading required package: nlme</span></span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="co">#&gt; Loading required package: parallel</span></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="co">#&gt;  **********   GAMLSS Version 5.1-6  **********</span></span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="co">#&gt; For more on GAMLSS look at http://www.gamlss.org/</span></span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="co">#&gt; Type gamlssNews() to see new features/changes/bug fixes.</span></span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="co">#&gt; Loading required package: data.table</span></span>
<span id="cb1-19"><a href="#cb1-19"></a></span>
<span id="cb1-20"><a href="#cb1-20"></a><span class="co"># example BAM file, baits, and restriction fragments</span></span>
<span id="cb1-21"><a href="#cb1-21"></a>bam &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;Bre80_2q35.bam&#39;</span>, <span class="dt">package =</span> <span class="st">&#39;chicane&#39;</span>);</span>
<span id="cb1-22"><a href="#cb1-22"></a>baits &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;2q35.bed&#39;</span>, <span class="dt">package =</span> <span class="st">&#39;chicane&#39;</span>);</span>
<span id="cb1-23"><a href="#cb1-23"></a>fragments &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;GRCh38_HindIII_chr2.bed.gz&#39;</span>, <span class="dt">package =</span> <span class="st">&#39;chicane&#39;</span>); <span class="co"># HindIII fragments on chromosome 2</span></span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb1-26"><a href="#cb1-26"></a>  chicane.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(</span>
<span id="cb1-27"><a href="#cb1-27"></a>    <span class="dt">bam =</span> bam,</span>
<span id="cb1-28"><a href="#cb1-28"></a>    <span class="dt">baits =</span> baits,</span>
<span id="cb1-29"><a href="#cb1-29"></a>    <span class="dt">fragments =</span> fragments</span>
<span id="cb1-30"><a href="#cb1-30"></a>    );</span>
<span id="cb1-31"><a href="#cb1-31"></a></span>
<span id="cb1-32"><a href="#cb1-32"></a>    <span class="kw">print</span>( chicane.results[ <span class="dv">1</span><span class="op">:</span><span class="dv">10</span> ] );</span>
<span id="cb1-33"><a href="#cb1-33"></a>}</span>
<span id="cb1-34"><a href="#cb1-34"></a><span class="co">#&gt; Taking input= as a system command (&#39;sh /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/bin/prepare_bam.sh -b /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/Bre80_2q35.bam -c /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/2q35.bed -f /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/GRCh38_HindIII_chr2.bed.gz -d /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rbuild1291a65ddc5e4/chicane/vignettes &#39;) and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</span></span>
<span id="cb1-35"><a href="#cb1-35"></a><span class="co">#&gt;                    target.id                  bait.id target.chr target.start</span></span>
<span id="cb1-36"><a href="#cb1-36"></a><span class="co">#&gt;  1: chr2:217491009-217492415 chr2:217035649-217042355       chr2    217491009</span></span>
<span id="cb1-37"><a href="#cb1-37"></a><span class="co">#&gt;  2: chr2:216650000-216651600 chr2:217430817-217437011       chr2    216650000</span></span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="co">#&gt;  3: chr2:217290048-217292530 chr2:217430817-217437011       chr2    217290048</span></span>
<span id="cb1-39"><a href="#cb1-39"></a><span class="co">#&gt;  4: chr2:216269087-216273140 chr2:217430817-217437011       chr2    216269087</span></span>
<span id="cb1-40"><a href="#cb1-40"></a><span class="co">#&gt;  5: chr2:216551051-216551968 chr2:217430817-217437011       chr2    216551051</span></span>
<span id="cb1-41"><a href="#cb1-41"></a><span class="co">#&gt;  6: chr2:216879929-216883553 chr2:217430817-217437011       chr2    216879929</span></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="co">#&gt;  7: chr2:216696003-216698549 chr2:217430817-217437011       chr2    216696003</span></span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="co">#&gt;  8: chr2:216687613-216696003 chr2:217430817-217437011       chr2    216687613</span></span>
<span id="cb1-44"><a href="#cb1-44"></a><span class="co">#&gt;  9: chr2:216906471-216908390 chr2:217430817-217437011       chr2    216906471</span></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="co">#&gt; 10: chr2:217162758-217166250 chr2:217430817-217437011       chr2    217162758</span></span>
<span id="cb1-46"><a href="#cb1-46"></a><span class="co">#&gt;     target.end bait.chr bait.start  bait.end bait.to.bait bait.trans.count</span></span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="co">#&gt;  1:  217492415     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb1-48"><a href="#cb1-48"></a><span class="co">#&gt;  2:  216651600     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="co">#&gt;  3:  217292530     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-50"><a href="#cb1-50"></a><span class="co">#&gt;  4:  216273140     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-51"><a href="#cb1-51"></a><span class="co">#&gt;  5:  216551968     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-52"><a href="#cb1-52"></a><span class="co">#&gt;  6:  216883553     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-53"><a href="#cb1-53"></a><span class="co">#&gt;  7:  216698549     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="co">#&gt;  8:  216696003     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-55"><a href="#cb1-55"></a><span class="co">#&gt;  9:  216908390     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-56"><a href="#cb1-56"></a><span class="co">#&gt; 10:  217166250     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb1-57"><a href="#cb1-57"></a><span class="co">#&gt;     target.trans.count  distance count expected      p.value   q.value</span></span>
<span id="cb1-58"><a href="#cb1-58"></a><span class="co">#&gt;  1:                  0  452710.0    28 6.222070 0.0001352420 0.2346449</span></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="co">#&gt;  2:                  0  783114.0    16 2.992273 0.0004798454 0.8172368</span></span>
<span id="cb1-60"><a href="#cb1-60"></a><span class="co">#&gt;  3:                  0  142625.0    33 8.889424 0.0010581410 0.8172368</span></span>
<span id="cb1-61"><a href="#cb1-61"></a><span class="co">#&gt;  4:                  0 1162800.5     8 2.101973 0.0019527803 0.8172368</span></span>
<span id="cb1-62"><a href="#cb1-62"></a><span class="co">#&gt;  5:                  0  882404.5    11 2.722403 0.0021130042 0.8172368</span></span>
<span id="cb1-63"><a href="#cb1-63"></a><span class="co">#&gt;  6:                  0  552173.0    11 3.822236 0.0036069810 0.8172368</span></span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="co">#&gt;  7:                  0  736638.0    16 4.162590 0.0043689364 0.8172368</span></span>
<span id="cb1-65"><a href="#cb1-65"></a><span class="co">#&gt;  8:                  0  742106.0    15 3.999764 0.0055330101 0.8172368</span></span>
<span id="cb1-66"><a href="#cb1-66"></a><span class="co">#&gt;  9:                  0  526483.5    11 4.150239 0.0063231596 0.8172368</span></span>
<span id="cb1-67"><a href="#cb1-67"></a><span class="co">#&gt; 10:                  0  269410.0    24 8.030272 0.0063767687 0.8172368</span></span></code></pre></div>
<p>The <code>chicane</code> function returns a <code>data.table</code> object sorted by q-value. Each row contains information about the target and bait fragments, and the observed and expected number of reads linking the two fragments.</p>
</div>
<div id="preprocessing" class="section level1">
<h1>Pre-processing data</h1>
<p>The <code>chicane</code> function can operate directly on BAM files. To convert the BAM file into a text file, R calls the shell script <code>prepare_bam.sh</code>. This script relies on <a href="http://bedtools.readthedocs.io/en/latest/index.html">bedtools</a> to select reads overlapping the captured fragments and identify the restriction fragments corresponding to each read. The reads are also filtered to remove those in which both reads map to the same fragment. The resulting data is read into a <a href="https://CRAN.R-project.org/package=data.table"><code>data.table</code></a> object, and the number of reads linking any two fragments is calculated.</p>
<p>For users that have a digested genome output from HiCUP, this needs to be converted to BED format to be compatible with bedtools. CHiCANE includes a helper function, <code>convert.hicup.digest.bed()</code>, which can be used to convert a digested genome in HiCUP format to a BED file of the restiction digest fragments. Any BED file containing the coordinates of restriction enzyme digested fragments of the genome build used according to the experiment are appropriate to be used with CHiCANE.</p>
<p>The third and final input required is a baitmap as a BED file containing the coordinates of the fragments which contain a bait in the capture step of the experiment. This can be obtained by conducting a <code>bedtools intersect</code> command on the fragment digest above and a BED file with the coordinates of the probes included on the array used in the capture step.</p>
<p>These three files can be supplied to the <code>chicane()</code> function directly as it will work as a wrapper function for preparing the data and interaction calling together. However, to speed up model fitting, it is possible to pre-process the data with the <code>prepare.data()</code> function and run <code>chicane</code> directly on the resulting data table. Furthermore, running in this step-wise manner is useful for those wanting to add custom covariates not already included in the data table or test variations of parameters and/or distibutions in the interaction calling. To first compare biological replicates using the <code>compare.replicates()</code> function, <code>prepare.data()</code> should be run on each replicate separately to assess concordance before deciding to merge them. This processing can take a while (~45 minutes for a 19GB BAM file on a commodity computer with 13GB RAM), and only needs to be done once for each BAM.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb2-2"><a href="#cb2-2"></a>  interaction.data &lt;-<span class="st"> </span><span class="kw">prepare.data</span>(</span>
<span id="cb2-3"><a href="#cb2-3"></a>    <span class="dt">bam =</span> bam,</span>
<span id="cb2-4"><a href="#cb2-4"></a>    <span class="dt">baits =</span> baits,</span>
<span id="cb2-5"><a href="#cb2-5"></a>    <span class="dt">fragments =</span> fragments</span>
<span id="cb2-6"><a href="#cb2-6"></a>    );</span>
<span id="cb2-7"><a href="#cb2-7"></a>}</span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co">#&gt; Taking input= as a system command (&#39;sh /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/bin/prepare_bam.sh -b /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/Bre80_2q35.bam -c /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/2q35.bed -f /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rinst1291a42eb2ffb/chicane/extdata/GRCh38_HindIII_chr2.bed.gz -d /private/var/folders/ch/h71q7lwx2wdb07ry2mlp337r000k4d/T/RtmpdNMaET/Rbuild1291a65ddc5e4/chicane/vignettes &#39;) and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</span></span></code></pre></div>
<p>The interaction data object contains details of the fragments, and the number of reads linking the two fragments.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) <span class="kw">print</span>(interaction.data);</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#&gt;                      target.id                  bait.id target.chr target.start</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co">#&gt;    1: chr2:200013520-200014510 chr2:217035649-217042355       chr2    200013520</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co">#&gt;    2: chr2:200018581-200020582 chr2:217035649-217042355       chr2    200018581</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">#&gt;    3: chr2:200020582-200023537 chr2:217430817-217437011       chr2    200020582</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">#&gt;    4: chr2:200033118-200034885 chr2:217430817-217437011       chr2    200033118</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">#&gt;    5: chr2:200040560-200041826 chr2:217430817-217437011       chr2    200040560</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">#&gt;   ---                                                                          </span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">#&gt; 4477: chr2:229867689-229872871 chr2:217035649-217042355       chr2    229867689</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">#&gt; 4478: chr2:229872871-229875614 chr2:217035649-217042355       chr2    229872871</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">#&gt; 4479: chr2:229882404-229887121 chr2:217430817-217437011       chr2    229882404</span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co">#&gt; 4480: chr2:229894321-229904945 chr2:217035649-217042355       chr2    229894321</span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="co">#&gt; 4481: chr2:229917543-229919836 chr2:217430817-217437011       chr2    229917543</span></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="co">#&gt;       target.end bait.chr bait.start  bait.end bait.to.bait bait.trans.count</span></span>
<span id="cb3-15"><a href="#cb3-15"></a><span class="co">#&gt;    1:  200014510     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb3-16"><a href="#cb3-16"></a><span class="co">#&gt;    2:  200020582     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="co">#&gt;    3:  200023537     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="co">#&gt;    4:  200034885     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb3-19"><a href="#cb3-19"></a><span class="co">#&gt;    5:  200041826     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb3-20"><a href="#cb3-20"></a><span class="co">#&gt;   ---                                                                       </span></span>
<span id="cb3-21"><a href="#cb3-21"></a><span class="co">#&gt; 4477:  229872871     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb3-22"><a href="#cb3-22"></a><span class="co">#&gt; 4478:  229875614     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb3-23"><a href="#cb3-23"></a><span class="co">#&gt; 4479:  229887121     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb3-24"><a href="#cb3-24"></a><span class="co">#&gt; 4480:  229904945     chr2  217035649 217042355        FALSE                0</span></span>
<span id="cb3-25"><a href="#cb3-25"></a><span class="co">#&gt; 4481:  229919836     chr2  217430817 217437011        FALSE                0</span></span>
<span id="cb3-26"><a href="#cb3-26"></a><span class="co">#&gt;       target.trans.count distance count</span></span>
<span id="cb3-27"><a href="#cb3-27"></a><span class="co">#&gt;    1:                  0 17024987     1</span></span>
<span id="cb3-28"><a href="#cb3-28"></a><span class="co">#&gt;    2:                  0 17019420     1</span></span>
<span id="cb3-29"><a href="#cb3-29"></a><span class="co">#&gt;    3:                  0 17411854     2</span></span>
<span id="cb3-30"><a href="#cb3-30"></a><span class="co">#&gt;    4:                  0 17399912     1</span></span>
<span id="cb3-31"><a href="#cb3-31"></a><span class="co">#&gt;    5:                  0 17392721     1</span></span>
<span id="cb3-32"><a href="#cb3-32"></a><span class="co">#&gt;   ---                                  </span></span>
<span id="cb3-33"><a href="#cb3-33"></a><span class="co">#&gt; 4477:                  0 12831278     1</span></span>
<span id="cb3-34"><a href="#cb3-34"></a><span class="co">#&gt; 4478:                  0 12835240     1</span></span>
<span id="cb3-35"><a href="#cb3-35"></a><span class="co">#&gt; 4479:                  0 12450848     1</span></span>
<span id="cb3-36"><a href="#cb3-36"></a><span class="co">#&gt; 4480:                  0 12860631     1</span></span>
<span id="cb3-37"><a href="#cb3-37"></a><span class="co">#&gt; 4481:                  0 12484776     1</span></span></code></pre></div>
<p>The <code>interactions</code> parameter of the <code>chicane()</code> function can take either a <code>data.table</code> object from <code>prepare.data</code> or the path to such a table.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb4-2"><a href="#cb4-2"></a>  file.name &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="st">&#39;interaction_data&#39;</span>);</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="kw">write.table</span>(interaction.data, file.name, <span class="dt">row.names =</span> <span class="ot">FALSE</span>);</span>
<span id="cb4-4"><a href="#cb4-4"></a>  chicane.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(<span class="dt">interactions =</span> file.name); </span>
<span id="cb4-5"><a href="#cb4-5"></a>}</span></code></pre></div>
</div>
<div id="statistical-model" class="section level1">
<h1>Statistical model</h1>
<p>CHiCANE models the expected number of paired reads containing two restriction fragments as a function of distance between them and how frequently the bait interacts with other fragments.</p>
<p>In the CHiCANE model, <span class="math inline">\(Y_{ij}\)</span> represents the number of reads linking bait <span class="math inline">\(i\)</span> and target fragment <span class="math inline">\(j\)</span>, such that it is assumed to follow a distribution with a mean <span class="math inline">\(\mu_{ij}\)</span> as:</p>
<p><span class="math display">\[\begin{equation}
    \mu_{ij} = \beta_0 + \beta_1\log(d_{ij}) + \beta_2\log(t_i + 1)
\end{equation}\]</span></p>
<p>where <span class="math inline">\(d_{ij}\)</span> represents the distance between the two fragments, and <span class="math inline">\(t_i\)</span> represents the number of reads linking the bait with a <span class="math inline">\(trans\)</span> fragment.</p>
<p>For bait to bait interactions, terms are added to the model to adjust for <span class="math inline">\(trans\)</span> counts of the other end, fragment <span class="math inline">\(j\)</span>, and the product of <span class="math inline">\(trans\)</span> counts of both fragments as:</p>
<p><span class="math display">\[\begin{equation}
    \mu_{ij} = \beta_0 + \beta_1\log(d_{ij}) + \beta_2\log(t_i + 1) + \beta_3\log(t_j + 1) + \beta_4\log(t_i + 1)\log(t_j + 1)
\end{equation}\]</span></p>
<p>Each possible interaction is assaigned an expected <span class="math inline">\(\mu_{ij}\)</span> and a <span class="math inline">\(p\)</span>-value for the observed counts <span class="math inline">\(y_{ij}\)</span> versus what is expected by chance is calculated as:</p>
<p><span class="math display">\[\begin{equation}
p = P(Y_{ij} \geq y_{ij})
\end{equation}\]</span></p>
<p>This probability is dependent on how the distribution is modelled.</p>
<div id="distribution-of-the-counts" class="section level2">
<h2>Distribution of the counts</h2>
<p>The CHiCANE method supports several different distributions of the counts <span class="math inline">\(Y_{ij}\)</span> conditional on the mean <span class="math inline">\(\mu_{ij}\)</span>. By default the counts are assumed to follow a negative binomial distribution with <span class="math inline">\(E(Y_{ij}) = exp (\mu_{ij})\)</span>. After fitting the model by a maximum likelihood over all possible interactions, the fitted model is used to estimate a <span class="math inline">\(p\)</span>-value for each pair. Poisson, truncated Poisson, or truncated negative binomial distributions are also supported and can override the default negative binomial by being passed as the <code>distribution</code> parameter.</p>
<div id="negative-binomial" class="section level3">
<h3>Negative binomial</h3>
<p>Negative binomial is the default distribution in CHiCANE as it is deemed more appropriate for modelling datasets with high variance in comparison to Poisson due to its inclusion of a term for overdispersion.</p>
<p>Using this default distribution, the counts are modelled as <span class="math inline">\(Y \sim NB(\mu, \theta)\)</span>, with mean <span class="math inline">\(\mu\)</span> and dispersion term <span class="math inline">\(\theta\)</span>. The variance of the counts conditional on the mean is then</p>
<p><span class="math display">\[\begin{equation}
Var(Y) = \mu + \frac{\mu^2}{\theta}
\end{equation}\]</span></p>
<p>This model implements the <code>glm.nb()</code> function in the MASS package.</p>
<p>If the model fails to converge due to a lack of overdispersion in the data <code>chicane()</code> will attempt to diagnose the error and if this is found to be the case a Poisson distribution will be implemented instead.</p>
</div>
<div id="poisson" class="section level3">
<h3>Poisson</h3>
<p>The Poisson distribution assumes that <span class="math inline">\(Y \sim \text{Poisson}(\mu)\)</span>, such that <span class="math inline">\(Var(Y) = \mu\)</span>. As the Poisson model does not allow variance of counts to exceed the mean, capture Hi-C counts usually possesses more variability than can be explained using this distribution, the result of which is false positives.</p>
</div>
<div id="truncated-negative-binomial" class="section level3">
<h3>Truncated negative binomial</h3>
<p>By default CHiCANE only includes observed interactions, rather than all possible interactions, <span class="math inline">\(i.e.\)</span> leaving out zero counts (see <a href="#include-zeros">Inclusion of Zeros</a>). One way to account for this is in the statistical model, by fitting a truncated negative binomial distribution in the <code>chicane()</code> call which implements the <a href="https://cran.r-project.org/package=gamlss.tr">GAMLSS R package</a> <span class="citation">(Stasinopoulos and Rigby 2016)</span>. It assigns <span class="math inline">\(P(Y = 0) = 0\)</span>, and assumes proportional probabilities to the negative binomial model for values <span class="math inline">\(y &gt; 0\)</span>.</p>
</div>
<div id="truncated-poisson" class="section level3">
<h3>Truncated poisson</h3>
<p>The truncated Poisson distribution is also supported, which behaves much like the truncated negative binomial model, but rather assumes the probabilities are proportional to Poisson probabilities for values <span class="math inline">\(y&gt;0\)</span>.</p>
<p>It is worth noting that specifying either of the truncated distributions will significantly slow down the model fitting.</p>
</div>
</div>
<div id="adjusting-for-covariates" class="section level2">
<h2>Adjusting for covariates</h2>
<p>CHiCANE allows users to specify additional covariates that can be added to the model with the <code>adjustment.terms</code> parameter. For example, we can add a term to adjust for the chromosome of the target fragment as shown below.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">data</span>(bre80);</span>
<span id="cb5-2"><a href="#cb5-2"></a>adjusted.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(</span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="dt">interactions =</span> bre80, </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="dt">adjustment.terms =</span> <span class="st">&#39;target.chr&#39;</span></span>
<span id="cb5-5"><a href="#cb5-5"></a>  );</span>
<span id="cb5-6"><a href="#cb5-6"></a></span>
<span id="cb5-7"><a href="#cb5-7"></a><span class="kw">print</span>( adjusted.results[ <span class="dv">1</span><span class="op">:</span><span class="dv">5</span> ] );</span>
<span id="cb5-8"><a href="#cb5-8"></a><span class="co">#&gt;                   target.id                  bait.id target.chr target.start</span></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co">#&gt; 1:    chr19:1155128-1228414 chr2:217035649-217042355      chr19      1155128</span></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co">#&gt; 2: chr1:117125276-117135137 chr2:217035649-217042355       chr1    117125276</span></span>
<span id="cb5-11"><a href="#cb5-11"></a><span class="co">#&gt; 3:   chr1:14950583-14951791 chr2:217035649-217042355       chr1     14950583</span></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co">#&gt; 4:   chr1:49076960-49080684 chr2:217035649-217042355       chr1     49076960</span></span>
<span id="cb5-13"><a href="#cb5-13"></a><span class="co">#&gt; 5: chr3:145680459-145682483 chr2:217035649-217042355       chr3    145680459</span></span>
<span id="cb5-14"><a href="#cb5-14"></a><span class="co">#&gt;    target.end bait.chr bait.start  bait.end bait.to.bait bait.trans.count</span></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co">#&gt; 1:    1228414     chr2  217035649 217042355        FALSE             9494</span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co">#&gt; 2:  117135137     chr2  217035649 217042355        FALSE             9494</span></span>
<span id="cb5-17"><a href="#cb5-17"></a><span class="co">#&gt; 3:   14951791     chr2  217035649 217042355        FALSE             9494</span></span>
<span id="cb5-18"><a href="#cb5-18"></a><span class="co">#&gt; 4:   49080684     chr2  217035649 217042355        FALSE             9494</span></span>
<span id="cb5-19"><a href="#cb5-19"></a><span class="co">#&gt; 5:  145682483     chr2  217035649 217042355        FALSE             9494</span></span>
<span id="cb5-20"><a href="#cb5-20"></a><span class="co">#&gt;    target.trans.count distance count expected   p.value   q.value</span></span>
<span id="cb5-21"><a href="#cb5-21"></a><span class="co">#&gt; 1:                  2       NA     2 1.003963 0.2656989 0.6417435</span></span>
<span id="cb5-22"><a href="#cb5-22"></a><span class="co">#&gt; 2:                  2       NA     2 1.003992 0.2657096 0.6417435</span></span>
<span id="cb5-23"><a href="#cb5-23"></a><span class="co">#&gt; 3:                  2       NA     2 1.003992 0.2657096 0.6417435</span></span>
<span id="cb5-24"><a href="#cb5-24"></a><span class="co">#&gt; 4:                  2       NA     2 1.003992 0.2657096 0.6417435</span></span>
<span id="cb5-25"><a href="#cb5-25"></a><span class="co">#&gt; 5:                  2       NA     2 1.004014 0.2657179 0.6417435</span></span></code></pre></div>
<p>The <code>adjustment.terms</code> parameter also supports expressions such as log transformations. Multiple adjustment terms can be added by passing as a vector.</p>
<div id="custom-covariates" class="section level3">
<h3>Custom covariates</h3>
<p>Any adjustment term must correspond to a column already present in the data. If a user would like to adjust for anything that is not already present in the CHiCANE output (<span class="math inline">\(e.g.\)</span> GC content), there’s a three step approach:</p>
<ol style="list-style-type: decimal">
<li>Run <code>prepare.data()</code> on your BAM file(s)</li>
<li>Add extra columns containing the desired covariate to the resulting data frame</li>
<li>Run <code>chicane()</code> by setting the <code>interactions</code> parameter to the output from step 2</li>
</ol>
</div>
</div>
<div id="filtering-baits-and-targets" class="section level2">
<h2>Filtering baits and targets</h2>
<p>By default, all baits and targets are included when fitting the CHiCANE model. An alternative approach is to filter out baits and fragments with low <span class="citation">(Dryden et al. 2014)</span> or high <span class="citation">(Cairns et al. 2016)</span> degree of “interactibility”, which is inferred via <span class="math inline">\(trans\)</span> counts for each bait. CHiCANE also supports this filtering through the <code>bait.filters</code> and <code>target.filters</code> parameters.</p>
<p>Each of these take a vector of length two, where each element corresponds to the proportion of fragments that should be considered to fall below the “lower limit” and “upper limit.” For example, passing in <code>bait.filters = c(0.3, 0.9)</code> means that the baits with the lowest 30% or highest 10% of <span class="math inline">\(trans\)</span> counts will be removed before fitting the model.</p>
<p>Filtering fragments may affect multiple testing correction by changing the number of tests performed.</p>
</div>
<div id="merging-biological-replicates" class="section level2">
<h2>Merging biological replicates</h2>
<p>The helper function <code>compare.replicates()</code> can be used to assess concordance between biological replicates before deciding whether to merge them to improve signal to noise ratio. This is done by conducting the <code>prepare.data()</code> step on each BAM separately before passing to the <code>compare.replicates()</code> function.</p>
<p>CHiCANE can merge replicates at the data processing step. If more than one BAM file is available, these can be passed as a vector to the <code>bam</code> parameter of the <code>chicane()</code> and <code>prepare.data()</code> functions. The <code>replicate.merging.method</code> parameter determines how replicates are merged. Available options are ‘sum’ and ‘weighted-sum’, with the former being the default.</p>
</div>
<div id="assessing-model-fits" class="section level2">
<h2>Assessing model fits</h2>
<p>In order to assess goodness of fit for the model implemented, CHiCANE outputs rootogram model fit plots and statistics to the directory provided through the <code>interim.data.directory</code> parameter in the <code>chicane()</code> function. The default <code>NULL</code> will skip these so it is important for a user to set this directory in order assess the model fit.</p>
</div>
<div id="setting-distance-bins" class="section level2">
<h2>Setting distance bins</h2>
<p>Adjusting for distance is done in a piecewise linear fashion. <span class="math inline">\(cis\)</span> data is ordered by distance, and split into equally sized bins. The count model is then fit separately within each bin, and results are merged at the end by concatenating the results from the individual model fits.</p>
<p>By default, CHiCANE will split the data into distance quantiles, <span class="math inline">\(i.e.\)</span> 100 equally sized groups. However, if the resulting datasets are too small, the number of distance bins is halved until all resulting datasets are large enough for the model to be fit. To override this default, pass the desired number of distance bins to <code>distance.bins</code>. For example, setting <code>distance.bins = 4</code> results in the count model being fit separately in each quartile. <span class="math inline">\(trans\)</span> and <span class="math inline">\(cis\)</span> interactions are fit separately.</p>
</div>
<div id="multiple-testing-correction" class="section level2">
<h2>Multiple testing correction</h2>
<p>Finally, CHiCANE applies the Benjamini-Hochberg method of multiple testing correction separately for each bait. To change this to a global multiple testing correction, set <code>multiple.testing.correction = &#39;global&#39;</code> in the main <code>chicane()</code> function.</p>
<p>The <code>chicane()</code> function’s final output is a q-value sorted data table of all interactions included in the model fitting. This full interactions table is useful in some instances such as the <a href="#locus-plot">locus plot</a> and <a href="#background-interactions">background interactions</a> functionality discussed below. However, its large size is rather unwieldy for most downstream analysis, so it is useful to also filter for only interactions below a desired q-value threshold and save this more workable table as well.</p>
</div>
</div>
<div id="interpretation-of-interaction-peaks" class="section level1">
<h1>Interpretation of interaction peaks</h1>
<p>In order to assist users with interpretation of interaction calls, CHiCANE also includes several helper functions for this purpose designed to be applicable to a myriad of experimental designs as delineated below and in more detail in our aforementioned manuscript.</p>
<div id="annotation-with-bedtools" class="section level2">
<h2>Annotation with bedtools</h2>
<p>The first step in interpretation of the interaction calls is likely to be annotation of the target fragments. Selecting just columns 3-5 from the CHiCANE interactions data table and saving the resulting coordinates as a BED file will then allow a <code>bedtools intersect</code> to be done between the target fragments and a genome annotation file also in BED format. The resulting output will indicate the genes that any target fragments overlap with.</p>
<p>Similarly, an intersect could be performed with VCF files containing variant calls or BED files containing ChIP-Seq peak calls for histone marks or CTCF binding sites. There are many disparate datasets which use a file type compatible with <code>bedtools intersect</code> and numerous options for the command to customise the output.</p>
</div>
<div id="visualising-interaction-peaks-in-washu-epigenome-browser" class="section level2">
<h2>Visualising interaction peaks in WashU Epigenome Browser</h2>
<p>CHiCANE also contains a function to convert its results into a format compatible with the WashU Epigenome Browser <span class="citation">(Li et al. 2019)</span>, <code>create.standard.format()</code>. The path to significant interaction calling results is provided to the <code>chicane.results</code> parameter and the desired name for the output file is passed to <code>file.name</code>. Once standard format is obtained it can be uploaded to the browser from the ‘Tracks’ dropdown, selecting ‘Local Text tracks’. In the upload window select ‘long-range format by CHiCANE’ as the text file type and then your standard format file in ‘Browse’. Once the track is loaded in the browser, you can right click on the track and set the Display Mode to ‘ARC’.</p>
</div>
<div id="locus-plot" class="section level2">
<h2>Locus plots</h2>
<p>CHiCANE results are also usable with the Gviz <span class="citation">(Hahne and Ivanek 2016)</span> package to visualise along with highly customisable genome anootation tracks. Additionally, the <code>create.locus.plot()</code> function in the CHiCANE package will make a basic implementation of such a plot with 6 tracks:</p>
<ol style="list-style-type: decimal">
<li>Ideogram of selected chromosome</li>
<li>Genome axis of the chromosome</li>
<li>Genome annotation provided by the user</li>
<li>A feature annotation displaying coordinates of any feature as provided by user in BED format (<span class="math inline">\(e.g.\)</span> baitmap, TADs, CTCF)</li>
<li>Raw counts of fragments from full interaction data table</li>
<li>Interactions below user-specified FDR threshold</li>
</ol>
<p>The user needs to provide the genome build to the <code>genome</code> parameter and coordinates of the locus desired to be plotted to the <code>chr</code>, <code>start</code>, and <code>end</code> parameters. A genome annotation file in GTF format needs to be provided to <code>gene.data</code>. The genomic feature chosen to be included will be passed as a BED format file to <code>genomic.features</code> and the desired name for this track is passed as <code>feature.name</code>. Finally, the path to the FULL interaction calls table is passed to <code>interaction.data</code> and the name for the output plot is set by <code>file.name</code>. By default only interactions below an FDR threshold of 0.05 will be included in the interactions track; a user can override this default by setting the `fdr.filter’ parameter. A brief example keeping all other parameters at default is shown below:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="kw">create.locus.plot</span>(</span>
<span id="cb6-3"><a href="#cb6-3"></a>    <span class="dt">genome =</span> <span class="st">&#39;hg38&#39;</span>,</span>
<span id="cb6-4"><a href="#cb6-4"></a>    <span class="dt">chr =</span> <span class="st">&#39;chr2&#39;</span>, <span class="dt">start =</span> <span class="dv">111100000</span>, <span class="dt">end =</span> <span class="dv">112200000</span>,</span>
<span id="cb6-5"><a href="#cb6-5"></a>    <span class="dt">gene.data =</span> <span class="st">&#39;/path/to/gene_data.gtf&#39;</span>,</span>
<span id="cb6-6"><a href="#cb6-6"></a>    <span class="dt">genomic.features =</span> <span class="st">&#39;/path/to/baitmap.bed&#39;</span>,</span>
<span id="cb6-7"><a href="#cb6-7"></a>    <span class="dt">feature.name =</span> <span class="st">&#39;baits&#39;</span>,</span>
<span id="cb6-8"><a href="#cb6-8"></a>    <span class="dt">interaction.data =</span> <span class="st">&#39;/path/to/interaction_calls.txt&#39;</span>,</span>
<span id="cb6-9"><a href="#cb6-9"></a>    <span class="dt">file.name =</span> <span class="st">&#39;/path/to/locus_plot.pdf&#39;</span></span>
<span id="cb6-10"><a href="#cb6-10"></a>    ); </span>
<span id="cb6-11"><a href="#cb6-11"></a>}</span></code></pre></div>
</div>
<div id="background-interactions" class="section level2">
<h2>Background Interactions</h2>
<p>In order to accurately estimate histone marker enrichment, the helper function <code>stratified.enrichment.sample()</code> will randomise interactions which match the number and proportion of observed interactions which serve as background for comparison, stratified by distance bins: 0-100kbp, 100kbp-1Mbp, 1Mbp-10Mbp, 10Mbp+, and <span class="math inline">\(trans\)</span>. Full paths to data tables containing significant and insignificant calls are passed to <code>significant.results</code> and <code>nonsignificant.results</code>, respectively.</p>
</div>
</div>
<div id="example-workflow-t-47d-breast-cancer-cell-line" class="section level1">
<h1>Example Workflow (T-47D breast cancer cell line)</h1>
<p>The code in this workflow is intended to be run on Baxter T-47D from the Zenodo directory of Supplementary Data for our manuscript at: <a href="https://zenodo.org/record/4073433" class="uri">https://zenodo.org/record/4073433</a></p>
<p>First pre-processing of input files needs to be completed as discussed above (see <a href="#preprocessing">Pre-Processing Data</a>). The BAM files referred to in this vignette have been created using HiCUP, the digested fragments were processed with the <code>convert.hicup.digest.bed()</code> function, and the baitmap was created via <code>bedtools intersect</code> between the aforementioned fragments and array bait probes used in the capture step. The fragments used here (hg38, HindIII digested) are found in Zenodo directory under <code>data/</code> and the baitmap is found in <code>data/captured_fragments/T47D/</code> as used in the <code>prepare.data()</code> call below.</p>
<p>The interactions data table can then be prepared with these processed inputs (aligned BAMs, fragments BED, and baitmap BED). The resulting output can then be saved before being passed to the main <code>chicane()</code> function for interaction calling. Once completed save both the full interaction calls results as well as calls filtered for significance to be used for different purposes.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb7-2"><a href="#cb7-2"></a></span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="co"># initial prep of interactions data table</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>  interaction.data &lt;-<span class="st"> </span><span class="kw">prepare.data</span>(</span>
<span id="cb7-5"><a href="#cb7-5"></a>    <span class="dt">bam =</span> <span class="kw">list.files</span>(<span class="st">&#39;data/bams/T47D&#39;</span>, <span class="dt">pattern =</span> <span class="st">&#39;.bam&#39;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>),</span>
<span id="cb7-6"><a href="#cb7-6"></a>    <span class="dt">baits =</span> <span class="st">&#39;data/captured_fragments/T47D/Baxter_captured_fragments.bed&#39;</span>,</span>
<span id="cb7-7"><a href="#cb7-7"></a>    <span class="dt">fragments =</span> <span class="st">&#39;data/GRCh38_HindIII_fragments.bed&#39;</span></span>
<span id="cb7-8"><a href="#cb7-8"></a>    );</span>
<span id="cb7-9"><a href="#cb7-9"></a></span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="co"># save interactions data table</span></span>
<span id="cb7-11"><a href="#cb7-11"></a>  file.name &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="st">&#39;interaction_data&#39;</span>);</span>
<span id="cb7-12"><a href="#cb7-12"></a>  <span class="kw">write.table</span>(interaction.data, file.name, <span class="dt">row.names =</span> <span class="ot">FALSE</span>);</span>
<span id="cb7-13"><a href="#cb7-13"></a></span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="co"># run interaction calling and save model fit plots and statistics</span></span>
<span id="cb7-15"><a href="#cb7-15"></a>  chicane.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(</span>
<span id="cb7-16"><a href="#cb7-16"></a>    <span class="dt">interactions =</span> file.name, </span>
<span id="cb7-17"><a href="#cb7-17"></a>    <span class="dt">interim.data.dir =</span> <span class="st">&#39;graphs/T47D/model_fits&#39;</span></span>
<span id="cb7-18"><a href="#cb7-18"></a>    ); </span>
<span id="cb7-19"><a href="#cb7-19"></a></span>
<span id="cb7-20"><a href="#cb7-20"></a>  <span class="co"># save full interaction calls table</span></span>
<span id="cb7-21"><a href="#cb7-21"></a>  <span class="kw">write.table</span>(</span>
<span id="cb7-22"><a href="#cb7-22"></a>    chicane.results, </span>
<span id="cb7-23"><a href="#cb7-23"></a>    <span class="dt">file =</span> <span class="st">&#39;data/chicane/T47D/interaction_calls.txt&#39;</span>, </span>
<span id="cb7-24"><a href="#cb7-24"></a>    <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </span>
<span id="cb7-25"><a href="#cb7-25"></a>    <span class="dt">quote =</span> <span class="ot">FALSE</span>, </span>
<span id="cb7-26"><a href="#cb7-26"></a>    <span class="dt">sep =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span></span>
<span id="cb7-27"><a href="#cb7-27"></a>    );</span>
<span id="cb7-28"><a href="#cb7-28"></a></span>
<span id="cb7-29"><a href="#cb7-29"></a>  <span class="co"># filter for significant only and save these results separately</span></span>
<span id="cb7-30"><a href="#cb7-30"></a>  significant.results &lt;-<span class="st"> </span>chicane.results[q.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>]</span>
<span id="cb7-31"><a href="#cb7-31"></a></span>
<span id="cb7-32"><a href="#cb7-32"></a>  <span class="kw">write.table</span>(</span>
<span id="cb7-33"><a href="#cb7-33"></a>    significant.results, </span>
<span id="cb7-34"><a href="#cb7-34"></a>    <span class="dt">file =</span> <span class="st">&#39;data/chicane/T47D/interaction_calls_significant.txt&#39;</span>, </span>
<span id="cb7-35"><a href="#cb7-35"></a>    <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </span>
<span id="cb7-36"><a href="#cb7-36"></a>    <span class="dt">quote =</span> <span class="ot">FALSE</span>, </span>
<span id="cb7-37"><a href="#cb7-37"></a>    <span class="dt">sep =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span></span>
<span id="cb7-38"><a href="#cb7-38"></a>    );</span>
<span id="cb7-39"><a href="#cb7-39"></a></span>
<span id="cb7-40"><a href="#cb7-40"></a>}</span></code></pre></div>
<p>After completion of interaction calling, the model fit plots and statistics in the <code>interim.data.dir</code> should be examined to ensure goodness of fit (detailed discussion of this in manuscript). Additionally, it is useful to examine the proportion of bait-to-bait, <span class="math inline">\(cis\)</span> and <span class="math inline">\(trans\)</span> interactions in the significant calls and the number of interaction calls by distance bins, which can be done with the following code:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb8-2"><a href="#cb8-2"></a></span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="co"># calculate proportions by interaction type</span></span>
<span id="cb8-4"><a href="#cb8-4"></a>  total &lt;-<span class="st"> </span><span class="kw">nrow</span>(significant.results);</span>
<span id="cb8-5"><a href="#cb8-5"></a>  trans.prop &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">is.na</span>(significant.results<span class="op">$</span>distance))<span class="op">/</span>total;</span>
<span id="cb8-6"><a href="#cb8-6"></a>  cis.prop &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(significant.results<span class="op">$</span>distance))<span class="op">/</span>total;</span>
<span id="cb8-7"><a href="#cb8-7"></a>  b2b.prop &lt;-<span class="st"> </span><span class="kw">sum</span>(significant.results<span class="op">$</span>bait.to.bait)<span class="op">/</span>total;</span>
<span id="cb8-8"><a href="#cb8-8"></a>  int.data &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;trans&#39;</span> =<span class="st"> </span>trans.prop, <span class="st">&#39;cis&#39;</span> =<span class="st"> </span>cis.prop, <span class="st">&#39;bait.to.bait&#39;</span> =<span class="st"> </span>b2b.prop);</span>
<span id="cb8-9"><a href="#cb8-9"></a>  <span class="kw">print</span>(int.data);</span>
<span id="cb8-10"><a href="#cb8-10"></a></span>
<span id="cb8-11"><a href="#cb8-11"></a>  <span class="co"># get number of interactions by distance bins</span></span>
<span id="cb8-12"><a href="#cb8-12"></a>  binned.data &lt;-<span class="st"> </span><span class="ot">NULL</span>;</span>
<span id="cb8-13"><a href="#cb8-13"></a></span>
<span id="cb8-14"><a href="#cb8-14"></a>  distance.bins &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb8-15"><a href="#cb8-15"></a>    <span class="st">&quot;0-10kbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">1e4</span>),</span>
<span id="cb8-16"><a href="#cb8-16"></a>    <span class="st">&quot;10-100kbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e4</span>, <span class="fl">1e5</span>),</span>
<span id="cb8-17"><a href="#cb8-17"></a>    <span class="st">&quot;100kbp-1Mbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e5</span>, <span class="fl">1e6</span>),</span>
<span id="cb8-18"><a href="#cb8-18"></a>    <span class="st">&quot;1-10Mbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e6</span>, <span class="fl">1e7</span>),</span>
<span id="cb8-19"><a href="#cb8-19"></a>    <span class="st">&quot;10-100Mbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e7</span>, <span class="fl">1e8</span>),</span>
<span id="cb8-20"><a href="#cb8-20"></a>    <span class="st">&quot;100-1000Mbp&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e8</span>, <span class="fl">1e9</span>)</span>
<span id="cb8-21"><a href="#cb8-21"></a>    );</span>
<span id="cb8-22"><a href="#cb8-22"></a></span>
<span id="cb8-23"><a href="#cb8-23"></a>  <span class="cf">for</span>(dist.i <span class="cf">in</span> <span class="kw">names</span>(distance.bins)){</span>
<span id="cb8-24"><a href="#cb8-24"></a>    bin.sum &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">which</span>(significant.results<span class="op">$</span>distance <span class="op">&gt;=</span><span class="st"> </span>distance.bins[[dist.i]][<span class="dv">1</span>] <span class="op">&amp;</span><span class="st"> </span>significant.results<span class="op">$</span>distance <span class="op">&lt;</span><span class="st"> </span>distance.bins[[dist.i]][<span class="dv">2</span>]));</span>
<span id="cb8-25"><a href="#cb8-25"></a>    binned.data &lt;-<span class="st"> </span><span class="kw">cbind</span>(binned.data, bin.sum);</span>
<span id="cb8-26"><a href="#cb8-26"></a>    }</span>
<span id="cb8-27"><a href="#cb8-27"></a></span>
<span id="cb8-28"><a href="#cb8-28"></a>  <span class="kw">colnames</span>(binned.data) &lt;-<span class="st"> </span><span class="kw">names</span>(distance.bins);</span>
<span id="cb8-29"><a href="#cb8-29"></a>  <span class="kw">print</span>(binned.data);</span>
<span id="cb8-30"><a href="#cb8-30"></a></span>
<span id="cb8-31"><a href="#cb8-31"></a>}</span></code></pre></div>
<p>Once the interaction calls are determined to be reasonable, it is useful to examine the significant interactions in context. Uploading the files to the WashU Epigenome Browser is a quick and user-friendly way to do so. Use the CHiCANE helper function to easily convert to the correct format for the browser.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb9-2"><a href="#cb9-2"></a></span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="co"># make a browser compatible file from significant interactions</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>  browser.file &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="st">&#39;significant_calls_standard.txt&#39;</span>)</span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="kw">create.standard.format</span>(</span>
<span id="cb9-6"><a href="#cb9-6"></a>    <span class="dt">chicane.results =</span> <span class="st">&#39;data/chicane/T47D/interaction_calls_significant.txt&#39;</span>,</span>
<span id="cb9-7"><a href="#cb9-7"></a>    <span class="dt">file.name =</span> browser.file</span>
<span id="cb9-8"><a href="#cb9-8"></a>    );</span>
<span id="cb9-9"><a href="#cb9-9"></a></span>
<span id="cb9-10"><a href="#cb9-10"></a>}</span></code></pre></div>
<p>Viewing in the browser will help to elucidate interaction peaks for loci of interest, then a user can make a plot of interactions at a particular locus using another of CHiCANE’s helper functions. In the following example we plot interactions at the 2q35 locus.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {</span>
<span id="cb10-2"><a href="#cb10-2"></a></span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="co"># set genome annotation file </span></span>
<span id="cb10-4"><a href="#cb10-4"></a>  gene.ann &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;gencode_2q35.gtf&#39;</span>, <span class="dt">package =</span> <span class="st">&#39;chicane&#39;</span>);</span>
<span id="cb10-5"><a href="#cb10-5"></a></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="co"># generate 2q35 locus plot</span></span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="kw">create.locus.plot</span>(</span>
<span id="cb10-8"><a href="#cb10-8"></a>    <span class="dt">genome =</span> <span class="st">&#39;hg38&#39;</span>,</span>
<span id="cb10-9"><a href="#cb10-9"></a>    <span class="dt">chr =</span> <span class="st">&#39;chr2&#39;</span>,</span>
<span id="cb10-10"><a href="#cb10-10"></a>    <span class="dt">start =</span> <span class="dv">216600000</span>,</span>
<span id="cb10-11"><a href="#cb10-11"></a>    <span class="dt">end =</span> <span class="dv">217140000</span>,</span>
<span id="cb10-12"><a href="#cb10-12"></a>    <span class="dt">gene.data =</span> gene.ann,</span>
<span id="cb10-13"><a href="#cb10-13"></a>    <span class="dt">genomic.features =</span> <span class="st">&#39;data/captured_fragments/T47D/Baxter_captured_fragments.bed&#39;</span>,</span>
<span id="cb10-14"><a href="#cb10-14"></a>    <span class="dt">feature.name =</span> <span class="st">&#39;baits&#39;</span>,</span>
<span id="cb10-15"><a href="#cb10-15"></a>    <span class="dt">interaction.data =</span> <span class="st">&#39;data/chicane/T47D/interaction_calls.txt&#39;</span>,</span>
<span id="cb10-16"><a href="#cb10-16"></a>    <span class="dt">file.name =</span> <span class="st">&#39;graphs/T47D/2q35.pdf&#39;</span></span>
<span id="cb10-17"><a href="#cb10-17"></a>    );</span>
<span id="cb10-18"><a href="#cb10-18"></a></span>
<span id="cb10-19"><a href="#cb10-19"></a>}</span></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">sessionInfo</span>();</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="co">#&gt; R version 3.6.1 (2019-07-05)</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co">#&gt; Running under: macOS Sierra 10.12.6</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co">#&gt; </span></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co">#&gt; </span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="co">#&gt; locale:</span></span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co">#&gt; [1] C/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8</span></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co">#&gt; </span></span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb11-14"><a href="#cb11-14"></a><span class="co">#&gt; [1] parallel  splines   stats     graphics  grDevices utils     datasets </span></span>
<span id="cb11-15"><a href="#cb11-15"></a><span class="co">#&gt; [8] methods   base     </span></span>
<span id="cb11-16"><a href="#cb11-16"></a><span class="co">#&gt; </span></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb11-18"><a href="#cb11-18"></a><span class="co">#&gt; [1] chicane_0.1.5     data.table_1.12.8 gamlss.tr_5.1-0   gamlss_5.1-6     </span></span>
<span id="cb11-19"><a href="#cb11-19"></a><span class="co">#&gt; [5] nlme_3.1-148      gamlss.data_5.1-4 gamlss.dist_5.1-6 MASS_7.3-51.6    </span></span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="co">#&gt; </span></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="co">#&gt;  [1] Rcpp_1.0.4.6         compiler_3.6.1       formatR_1.7         </span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="co">#&gt;  [4] bedr_1.0.7           futile.logger_1.4.3  iterators_1.0.12    </span></span>
<span id="cb11-24"><a href="#cb11-24"></a><span class="co">#&gt;  [7] R.methodsS3_1.8.0    futile.options_1.0.1 R.utils_2.9.2       </span></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="co">#&gt; [10] tools_3.6.1          testthat_2.3.2       digest_0.6.25       </span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="co">#&gt; [13] evaluate_0.14        lattice_0.20-41      rlang_0.4.9         </span></span>
<span id="cb11-27"><a href="#cb11-27"></a><span class="co">#&gt; [16] Matrix_1.2-18        foreach_1.5.0        yaml_2.2.1          </span></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="co">#&gt; [19] VennDiagram_1.6.20   xfun_0.14            stringr_1.4.0       </span></span>
<span id="cb11-29"><a href="#cb11-29"></a><span class="co">#&gt; [22] knitr_1.28           grid_3.6.1           R6_2.4.1            </span></span>
<span id="cb11-30"><a href="#cb11-30"></a><span class="co">#&gt; [25] survival_3.1-12      rmarkdown_2.1        lambda.r_1.2.4      </span></span>
<span id="cb11-31"><a href="#cb11-31"></a><span class="co">#&gt; [28] magrittr_1.5         codetools_0.2-16     htmltools_0.4.0     </span></span>
<span id="cb11-32"><a href="#cb11-32"></a><span class="co">#&gt; [31] stringi_1.4.6        R.oo_1.23.0</span></span></code></pre></div>
</div>
<div id="acknowledgements" class="section level1">
<h1>Acknowledgements</h1>
<p>Development of CHiCANE was supported by <a href="https://breastcancernow.org:443/">Breast Cancer Now</a>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-baxter2018">
<p>Baxter, Joseph S, Olivia C Leavy, Nicola H Dryden, Sarah Maguire, Nichola Johnson, Vita Fedele, Nikiana Simigdala, et al. 2018. “Capture Hi-c Identifies Putative Target Genes at 33 Breast Cancer Risk Loci.” <em>Nature Communications</em> 9 (1): 1028.</p>
</div>
<div id="ref-cairns2016">
<p>Cairns, Jonathan, Paula Freire-Pritchett, Steven W Wingett, Csilla Várnai, Andrew Dimond, Vincent Plagnol, Daniel Zerbino, et al. 2016. “CHiCAGO: Robust Detection of Dna Looping Interactions in Capture Hi-c Data.” <em>Genome Biology</em> 17 (1): 127.</p>
</div>
<div id="ref-dryden2014">
<p>Dryden, Nicola H, Laura R Broome, Frank Dudbridge, Nichola Johnson, Nick Orr, Stefan Schoenfelder, Takashi Nagano, et al. 2014. “Unbiased Analysis of Potential Targets of Breast Cancer Susceptibility Loci by Capture Hi-c.” <em>Genome Research</em> 24 (11): 1854–68.</p>
</div>
<div id="ref-gviz">
<p>Hahne, Florian, and Robert Ivanek. 2016. “Statistical Genomics: Methods and Protocols.” In, edited by Ewy Mathé and Sean Davis, 335–51. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-1-4939-3578-9_16">https://doi.org/10.1007/978-1-4939-3578-9_16</a>.</p>
</div>
<div id="ref-10.1093/nar/gkz348">
<p>Li, Daofeng, Silas Hsu, Deepak Purushotham, Renee L Sears, and Ting Wang. 2019. “WashU Epigenome Browser update 2019.” <em>Nucleic Acids Research</em> 47 (W1): W158–W165. <a href="https://doi.org/10.1093/nar/gkz348">https://doi.org/10.1093/nar/gkz348</a>.</p>
</div>
<div id="ref-stasinopoulos2016">
<p>Stasinopoulos, Mikis, and Bob Rigby. 2016. <em>Gamlss.tr: Generating and Fitting Truncated “Gamlss.family” Distributions</em>. <a href="https://CRAN.R-project.org/package=gamlss.tr">https://CRAN.R-project.org/package=gamlss.tr</a>.</p>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
