<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">


<meta name="date" content="2019-09-29" />

<title>Getting Started with Capture Hi-C Analysis Engine</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Getting Started with Capture Hi-C Analysis Engine</h1>
<h4 class="date">2019-09-29</h4>


<div id="TOC">
<ul>
<li><a href="#quickstart">Quickstart</a></li>
<li><a href="#preprocessing">Pre-Processing Data</a></li>
<li><a href="#statistical-model">Statistical Model</a><ul>
<li><a href="#distribution-of-the-counts">Distribution of the Counts</a><ul>
<li><a href="#negative-binomial">Negative Binomial</a></li>
<li><a href="#poisson">Poisson</a></li>
<li><a href="#truncated-negative-binomial">Truncated Negative Binomial</a></li>
<li><a href="#truncated-poisson">Truncated Poisson</a></li>
</ul></li>
<li><a href="#adjusting-for-covariates">Adjusting for Covariates</a></li>
<li><a href="#filtering-baits-and-targets">Filtering Baits and Targets</a></li>
<li><a href="#multiple-testing-correction">Multiple Testing Correction</a></li>
<li><a href="#merging-biological-replicates">Merging Biological Replicates</a></li>
<li><a href="#adjusting-for-custom-covariates">Adjusting for Custom Covariates</a></li>
</ul></li>
<li><a href="#helper-functions">Helper Functions</a></li>
<li><a href="#session-info">Session Info</a></li>
<li><a href="#acknowledgements">Acknowledgements</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="quickstart" class="section level1">
<h1>Quickstart</h1>
<p>The main function for running the CHiCANE method is <code>chicane()</code>. This takes a bam file and bed files with all restriction fragments and targeted restriction fragments as input, and produces a table with fragment interactions and associated p-values.</p>
<p>The package comes with a subset of reads from a capture Hi-C experiment on the Bre80 normal epithelial breast tissue cell line <span class="citation">(Baxter et al. 2018)</span>. The experiment targeted several breast cancer risk loci in non-coding regions of the genome, and aimed to associate these risk variants with genes. Two of the risk SNPs are rs13387042 and rs16857609, and reads that map to them have been included in the file Bre80_2q35.bam.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(chicane);
<span class="co">#&gt; Loading required package: gamlss.tr</span>
<span class="co">#&gt; Loading required package: gamlss.dist</span>
<span class="co">#&gt; Loading required package: MASS</span>
<span class="co">#&gt; Loading required package: gamlss</span>
<span class="co">#&gt; Loading required package: splines</span>
<span class="co">#&gt; Loading required package: gamlss.data</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'gamlss.data'</span>
<span class="co">#&gt; The following object is masked from 'package:datasets':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     sleep</span>
<span class="co">#&gt; Loading required package: nlme</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt;  **********   GAMLSS Version 5.1-4  **********</span>
<span class="co">#&gt; For more on GAMLSS look at http://www.gamlss.org/</span>
<span class="co">#&gt; Type gamlssNews() to see new features/changes/bug fixes.</span>

<span class="co"># example BAM file, baits, and restriction fragments</span>
bam &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="st">'Bre80_2q35.bam'</span>, <span class="dt">package =</span> <span class="st">'chicane'</span>);
baits &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="st">'2q35.bed'</span>, <span class="dt">package =</span> <span class="st">'chicane'</span>);
fragments &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="st">'GRCh38_HindIII_chr2.bed.gz'</span>, <span class="dt">package =</span> <span class="st">'chicane'</span>); <span class="co"># HindIII fragments on chromosome 2</span>

<span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {
  chicane.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(
    <span class="dt">bam =</span> bam,
    <span class="dt">baits =</span> baits,
    <span class="dt">fragments =</span> fragments
    );

    <span class="kw">print</span>( chicane.results[ <span class="dv">1</span><span class="op">:</span><span class="dv">10</span> ] );
}</code></pre></div>
<p>The <code>chicane</code> function returns a <code>data.table</code> object sorted by q-value. Each row contains information about the target and bait fragments, and the observed and expected number of reads linking the two fragments.</p>
</div>
<div id="preprocessing" class="section level1">
<h1>Pre-Processing Data</h1>
<p>The <code>chicane</code> function can operate directly on BAM files. To convert the BAM file into a text file, R calls the shell script <code>prepare_bam.sh</code>. This script relies on <a href="http://bedtools.readthedocs.io/en/latest/index.html">bedtools</a> to select reads overlapping the captured fragments and identify the restriction fragments corresponding to each read. The resulting data is read into a <a href="https://CRAN.R-project.org/package=data.table"><code>data.table</code></a> object, and the number of reads linking any two fragments is calculated.</p>
<p>This processing can take a while (~45 minutes for a 19GB BAM file on a commodity computer with 13GB RAM), and only needs to be done once for each BAM. To speed up model fitting, it is possible to pre-process the data with the <code>prepare.data()</code> function and run <code>chicane</code> directly on the resulting data table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {
  interaction.data &lt;-<span class="st"> </span><span class="kw">prepare.data</span>(
    <span class="dt">bam =</span> bam,
    <span class="dt">baits =</span> baits,
    <span class="dt">fragments =</span> fragments
    );
}</code></pre></div>
<p>The interaction data object contains details of the fragments, and the number of reads linking the two fragments.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) <span class="kw">print</span>(interaction.data);</code></pre></div>
<p>By default, only combinations of fragments that are detected at least once are included in the data. To also include zero counts, use the <code>include.zeros</code> argument. Setting <code>include.zeros = 'cis'</code> includes all zero counts for bait/ target combinations on the same chromosome, while <code>include.zeros = 'all'</code> includes all possible combinations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {
  cis.zero.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(
    <span class="dt">bam =</span> bam,
    <span class="dt">baits =</span> baits,
    <span class="dt">fragments =</span> fragments,
    <span class="dt">include.zeros =</span> <span class="st">'cis'</span>
    ); 

    <span class="kw">print</span>( cis.zero.results[ <span class="dv">1</span><span class="op">:</span><span class="dv">10</span> ] );
}</code></pre></div>
<p>For most experiments including zeros will result in impractically large files. Another option for accounting for zero counts is to specify a truncated distribution through the <code>distribution</code> argument. However, this will slow down model fitting.</p>
<p>The <code>interactions</code> argument of the <code>chicane()</code> function can take either a <code>data.table</code> object from <code>prepare.data</code> or the path to such a table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>( <span class="kw">bedtools.installed</span>() ) {
  file.name &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="st">'interaction_data'</span>);
  <span class="kw">write.table</span>(interaction.data, file.name, <span class="dt">row.names =</span> <span class="ot">FALSE</span>);

  chicane.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(<span class="dt">interactions =</span> file.name); 
}</code></pre></div>
</div>
<div id="statistical-model" class="section level1">
<h1>Statistical Model</h1>
<p>The CHiCANE method models the expected number of reads linking two restriction fragments as a function of the distance between the loci and the “interactibility” of the bait fragment, that is, its inherent propensity to interact with other fragments.</p>
<p>Let <span class="math inline">\(Y_{ij}\)</span> denote the number of reads linking bait <span class="math inline">\(i\)</span> and target fragment <span class="math inline">\(j\)</span>, <span class="math inline">\(d_{ij}\)</span> be the distance between the two fragments, and <span class="math inline">\(t_i\)</span> denote the number of reads linking the bait <span class="math inline">\(i\)</span> with another fragment in trans. The CHiCANE model assumes that <span class="math inline">\(Y_{ij}\)</span> follows a distribution with mean parameter <span class="math inline">\(\mu_{ij}\)</span>, where</p>
<span class="math display">\[\begin{equation}
    \mu_{ij} = \beta_0 + \beta_1\log(d_{ij}) + \beta_2\log(t_i + 1)
\end{equation}\]</span>
<p>If the other end fragment <span class="math inline">\(j\)</span> is also a bait, terms are included to adjust for the trans counts of the fragment <span class="math inline">\(j\)</span>, as well as the product of the trans counts of the two fragments as shown below</p>
<span class="math display">\[\begin{equation}
    \mu_{ij} = \beta_0 + \beta_1\log(d_{ij}) + \beta_2\log(t_i + 1) + \beta_3\log(t_j + 1) + \beta_4\log(t_i + 1)\log(t_j + 1)
\end{equation}\]</span>
<p>Once an expected value <span class="math inline">\(\mu_{ij}\)</span> is obtained for each combination of fragments, a <span class="math inline">\(p\)</span>-value for the observed counts <span class="math inline">\(y_{ij}\)</span> exceeding what’s expected by random chance is calculated as</p>
<span class="math display">\[\begin{equation}
p = P(Y_{ij} \geq y_{ij})
\end{equation}\]</span>
<p>The probability <span class="math inline">\(P(Y_{ij} \geq y_{ij})\)</span> depends on how the distribution <span class="math inline">\(Y_{ij} \mid \mu_{ij}\)</span> of the counts conditional on the expected value is modelled.</p>
<div id="distribution-of-the-counts" class="section level2">
<h2>Distribution of the Counts</h2>
<p>The CHiCANE method supports several different distributions of the counts <span class="math inline">\(Y_{ij}\)</span> conditional on the mean <span class="math inline">\(\mu_{ij}\)</span>. By default the counts are assumed to follow a negative binomial distribution with <span class="math inline">\(E(Y_{ij}) = exp (\mu_{ij})\)</span>. The negative binomial is similar to the Poisson model for counts, but includes an overdispersion term that makes it more appropriate for datasets with high variance. After fitting <span class="math inline">\(\beta_0, \beta_1, \beta_2\)</span> and the overdispersion term by a maximum likelihood over all pairs of fragments, we use the fitted model to estimate a <span class="math inline">\(p\)</span>-value for each pair. The Poisson, truncated Poisson, or truncated negative binomial distribution can be specified through the <code>distribution</code> argument.</p>
<div id="negative-binomial" class="section level3">
<h3>Negative Binomial</h3>
<p>The negative binomial distribution models the counts as <span class="math inline">\(Y \sim NB(\mu, \theta)\)</span>, where <span class="math inline">\(\mu\)</span> is the mean and <span class="math inline">\(\theta\)</span> is a dispersion parameter. Under this model, the variance of the counts conditional on the mean <span class="math inline">\(\mu\)</span> is given by</p>
<span class="math display">\[\begin{equation}
Var(Y) = \mu + \frac{\mu^2}{\theta}
\end{equation}\]</span>
<p>The model is fit using the <code>glm.nb()</code> function in the MASS package. The maximum likelihood estimates (MLEs) of the coefficients <span class="math inline">\(\hat{\beta_i}\)</span> are used to provide an estimate of the expected number of counts for each combination of restriction fragments. The MLE of the dispersion parameter <span class="math inline">\(\theta\)</span> is used as the dispersion parameter (<code>size</code>) in the negative binomial model.</p>
<p>Occasionally, the model fitting procedure will result in numerical errors as the result of lack of convergence. This is most often due to lack of overdispersion in the data leading to an unstable estimate of <span class="math inline">\(\theta\)</span>. If the variance is approximately equal to the mean, the <span class="math inline">\(\theta\)</span> term goes to infinity and the model fails to converge.</p>
<p>When the model throws a numerical warning or fails to converge, the <code>chicane()</code> function will attempt to diagnose if the problem was due to lack of overdispersion. This is done by fitting a negative binomial regerssion model with 25 iterations, and using the resulting fitted values and dispersion parameter to examine the variance of the negative binomial. If the estimated variance at the point with the largest fitted value is less than 0.1% greater than the corresponding estimate of the Poisson variance (i.e. the fitted value at that point), a Poisson distribution is fit instead.</p>
</div>
<div id="poisson" class="section level3">
<h3>Poisson</h3>
<p>Unlike the negative binomial, the Poisson distribution does not allow for the variance of the counts to exceed their mean. This model assumes that <span class="math inline">\(Y \sim \text{Poisson}(\mu)\)</span>, giving <span class="math inline">\(Var(Y) = \mu\)</span>. In practice, capture Hi-C counts often show more variability than can be explained by the Poisson model, leading to false positive interaction calls.</p>
</div>
<div id="truncated-negative-binomial" class="section level3">
<h3>Truncated Negative Binomial</h3>
<p>When fitting the model without including unobserved combinations of fragments (see <a href="#preprocessing">Pre-processing Data</a>), there are no counts of zero in the data. To accommodate this in the statistical model, a truncated Poisson distribution can be fit using the <a href="https://cran.r-project.org/package=gamlss.tr">GAMLSS package</a> <span class="citation">(Stasinopoulos and Rigby 2016)</span>. This model assigns <span class="math inline">\(P(Y = 0) = 0\)</span>, and assumes the probabilities are proportional to the negative binomial probabilities for values <span class="math inline">\(y &gt; 0\)</span>.</p>
</div>
<div id="truncated-poisson" class="section level3">
<h3>Truncated Poisson</h3>
<p>Similarly to the truncated negative binomial, the truncated Poisson distribution is supported by setting <code>distribution = 'truncated-poisson'</code>. This model assumes the probabilities are proportional to the Poisson probabilities for values <span class="math inline">\(y&gt;0\)</span>.</p>
</div>
</div>
<div id="adjusting-for-covariates" class="section level2">
<h2>Adjusting for Covariates</h2>
<p>Adjusting for distance is done in a piecewise linear fashion. Cis-data is ordered by distance, and split into equally sized bins. The count model is then fit separately within each bin, and results are merged at the end by concatenating the results from the individual model fits.</p>
<p>By default, CHiCANE tries to split the data based on distance quantiles, i.e. into 100 equally sized groups. In some cases, particularly when dealing with smaller datasets, the resulting datasets are too small for model fitting. When this occurs, the number of distance bins are halved until all resulting datasets are large enough for the model to be fit. To override the default behaviour, pass the desired number of distance bins to <code>distance.bins</code>. For example, setting <code>distance.bins = 4</code> results in the count model being fit separately in each quartile. Trans interactions are fit separately from cis interactions.</p>
</div>
<div id="filtering-baits-and-targets" class="section level2">
<h2>Filtering Baits and Targets</h2>
<p>By default, all baits and targets are included when fitting the CHiCANE model. An alternative approach is to filter out baits and fragments with low <span class="citation">(Dryden et al. 2014)</span> or high <span class="citation">(Cairns et al. 2016)</span> degree of “interactibility”, as inferred through the trans counts. CHiCANE also supports this model through the <code>bait.filters</code> and <code>target.filters</code> arguments.</p>
<p>Each of these take a vector of length two, where each element corresponds to the proportion of fragments that should be considered to fall below the “lower limit” and “upper limit.” For example, passing in <code>bait.filters = c(0.3, 0.9)</code> means that the baits with the lowest 30% or highest 10% of trans counts will be removed before fitting the model.</p>
<p>Filtering fragments may affect multiple testing correction by changing the number of tests performed.</p>
</div>
<div id="multiple-testing-correction" class="section level2">
<h2>Multiple Testing Correction</h2>
<p>By default, CHiCANE performs multiple testing correction separately for each bait. To change this to a global multiple testing correction, set <code>multiple.testing.correction = 'global'</code> in the main <code>chicane()</code> function.</p>
</div>
<div id="merging-biological-replicates" class="section level2">
<h2>Merging Biological Replicates</h2>
<p>CHiCANE can merge replicates at the data processing step. If more than one BAM file is available, these can be passed as vector to the <code>bam</code> argument of the <code>chicane()</code> and <code>prepare.data()</code> functions. The <code>replicate.merging.method</code> determines how replicates are merged. Available options are ‘sum’ and ‘weighted-sum’.</p>
</div>
<div id="adjusting-for-custom-covariates" class="section level2">
<h2>Adjusting for Custom Covariates</h2>
<p>CHiCANE allows users to specify additional covariates that should be added to the model with the <code>adjustment.terms</code> argument. For example, we can add a term to adjust for the chromosome of the target fragment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(bre80);
adjusted.results &lt;-<span class="st"> </span><span class="kw">chicane</span>(
  <span class="dt">interactions =</span> bre80, 
  <span class="dt">adjustment.terms =</span> <span class="st">'target.chr'</span>
  );

<span class="kw">print</span>( adjusted.results[ <span class="dv">1</span><span class="op">:</span><span class="dv">5</span> ] );
<span class="co">#&gt;                   target.id                  bait.id target.chr</span>
<span class="co">#&gt; 1:    chr19:1155128-1228414 chr2:217035649-217042355      chr19</span>
<span class="co">#&gt; 2: chr1:117125276-117135137 chr2:217035649-217042355       chr1</span>
<span class="co">#&gt; 3:   chr1:14950583-14951791 chr2:217035649-217042355       chr1</span>
<span class="co">#&gt; 4:   chr1:49076960-49080684 chr2:217035649-217042355       chr1</span>
<span class="co">#&gt; 5: chr3:145680459-145682483 chr2:217035649-217042355       chr3</span>
<span class="co">#&gt;    target.start target.end bait.chr bait.start  bait.end bait.to.bait</span>
<span class="co">#&gt; 1:      1155128    1228414     chr2  217035649 217042355        FALSE</span>
<span class="co">#&gt; 2:    117125276  117135137     chr2  217035649 217042355        FALSE</span>
<span class="co">#&gt; 3:     14950583   14951791     chr2  217035649 217042355        FALSE</span>
<span class="co">#&gt; 4:     49076960   49080684     chr2  217035649 217042355        FALSE</span>
<span class="co">#&gt; 5:    145680459  145682483     chr2  217035649 217042355        FALSE</span>
<span class="co">#&gt;    bait.trans.count target.trans.count distance count expected   p.value</span>
<span class="co">#&gt; 1:             9494                  2       NA     2 1.003963 0.2656989</span>
<span class="co">#&gt; 2:             9494                  2       NA     2 1.003992 0.2657096</span>
<span class="co">#&gt; 3:             9494                  2       NA     2 1.003992 0.2657096</span>
<span class="co">#&gt; 4:             9494                  2       NA     2 1.003992 0.2657096</span>
<span class="co">#&gt; 5:             9494                  2       NA     2 1.004014 0.2657179</span>
<span class="co">#&gt;      q.value</span>
<span class="co">#&gt; 1: 0.6417435</span>
<span class="co">#&gt; 2: 0.6417435</span>
<span class="co">#&gt; 3: 0.6417435</span>
<span class="co">#&gt; 4: 0.6417435</span>
<span class="co">#&gt; 5: 0.6417435</span></code></pre></div>
<p>The <code>adjustment.terms</code> argument also supports more complex expressions such as log-transformations. Multiple adjustment terms can be added by passing a vector.</p>
<p>Any adjustment term must correspond to a column already present in the data. If you would like to adjust for anything that is not already present in the CHiCANE output (e.g. GC content), there’s a three step approach:</p>
<ol style="list-style-type: decimal">
<li>Run <code>prepare.data()</code> on your BAM file</li>
<li>Add extra columns to the resulting data frame</li>
<li>Run <code>chicane()</code> with the <code>interactions</code> argument</li>
</ol>
</div>
</div>
<div id="helper-functions" class="section level1">
<h1>Helper Functions</h1>
<p>The <code>convert.hicup.digest.bed()</code> function can be used to convert a digested genome in HiCUP format to a BED file.</p>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>();
<span class="co">#&gt; R Under development (unstable) (2019-09-27 r77229)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin16.7.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Sierra 10.12.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Users/shaider/Desktop/R-devel/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Users/shaider/Desktop/R-devel/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] C/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  splines   stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] data.table_1.12.2 chicane_0.1.1     gamlss.tr_5.1-0   gamlss_5.1-4     </span>
<span class="co">#&gt; [5] nlme_3.1-141      gamlss.data_5.1-4 gamlss.dist_5.1-4 MASS_7.3-51.4    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.2        knitr_1.25        magrittr_1.5     </span>
<span class="co">#&gt;  [4] lattice_0.20-38   foreach_1.4.7     stringr_1.4.0    </span>
<span class="co">#&gt;  [7] tools_3.7.0       grid_3.7.0        xfun_0.9         </span>
<span class="co">#&gt; [10] iterators_1.0.12  htmltools_0.3.6   yaml_2.2.0       </span>
<span class="co">#&gt; [13] survival_2.44-1.1 digest_0.6.21     Matrix_1.2-17    </span>
<span class="co">#&gt; [16] codetools_0.2-16  evaluate_0.14     rmarkdown_1.15   </span>
<span class="co">#&gt; [19] stringi_1.4.3     compiler_3.7.0</span></code></pre></div>
</div>
<div id="acknowledgements" class="section level1">
<h1>Acknowledgements</h1>
<p>Development of CHiCANE was supported by <a href="http://breastcancernow.org">Breast Cancer Now</a>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-baxter2018">
<p>Baxter, Joseph S, Olivia C Leavy, Nicola H Dryden, Sarah Maguire, Nichola Johnson, Vita Fedele, Nikiana Simigdala, et al. 2018. “Capture Hi-c Identifies Putative Target Genes at 33 Breast Cancer Risk Loci.” <em>Nature Communications</em> 9 (1). Nature Publishing Group: 1028.</p>
</div>
<div id="ref-cairns2016">
<p>Cairns, Jonathan, Paula Freire-Pritchett, Steven W Wingett, Csilla Várnai, Andrew Dimond, Vincent Plagnol, Daniel Zerbino, et al. 2016. “CHiCAGO: Robust Detection of Dna Looping Interactions in Capture Hi-c Data.” <em>Genome Biology</em> 17 (1). BioMed Central: 127.</p>
</div>
<div id="ref-dryden2014">
<p>Dryden, Nicola H, Laura R Broome, Frank Dudbridge, Nichola Johnson, Nick Orr, Stefan Schoenfelder, Takashi Nagano, et al. 2014. “Unbiased Analysis of Potential Targets of Breast Cancer Susceptibility Loci by Capture Hi-c.” <em>Genome Research</em> 24 (11). Cold Spring Harbor Lab: 1854–68.</p>
</div>
<div id="ref-stasinopoulos2016">
<p>Stasinopoulos, Mikis, and Bob Rigby. 2016. <em>Gamlss.tr: Generating and Fitting Truncated “Gamlss.family” Distributions</em>. <a href="https://CRAN.R-project.org/package=gamlss.tr" class="uri">https://CRAN.R-project.org/package=gamlss.tr</a>.</p>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
